src/d3/xy-chart.d3.ts

```diff
@@ -2,14 +2,12 @@ import * as d3 from 'd3';
 import { BaseD3, RenderOptions as BaseRenderOptions } from './base.d3';
 import { Observable } from 'rxjs';
 import { formatX } from '../utils/formatters';
-import { XYChartData, XYPoint } from '../datasets/types';
-
-// TODO: Generic Datum type
-export type DNPoint = XYPoint<Date, number>;
+import { TimeSeriesPoint } from '../datasets/queries/time-series.query';
+import { LineChartDatum } from '../components/line-chart/line-chart.component';
 
 export interface RenderOptions extends BaseRenderOptions {
-  data$: Observable<XYChartData>;
-  activeDatum$: Observable<DNPoint | null>;
+  data$: Observable<LineChartDatum>;
```

Comment:
My general advise for naming variables in this project:

`point` for [x, y] pair data
`points` for list of point (because data points)
`data` for object of points and other related information (and use `data.points` to get list of point to plot)
and please align variable names with their types. A variable with `data` in its name should be in `Data` type.

Although "datum" is the singular form of "data", the later is used more often and are usually even used in both cases. I would suggest to use "data" whenever possible in this project.

Reply:
I agree on "point" and "points". It is true that `data` is much more commonly used as a collective noun, but I would still lean on calling it "datum" so that we can address `Datum[]` as `Data`. (i.e., naming it `datas` is grammatically wrong.)

I'll align the variable names with their types in #45 also, to prevent massive merge conflicts.